models/pull.go

```diff
@@ -1037,6 +1037,21 @@ func GetUnmergedPullRequestsByHeadInfo(repoID int64, branch string) ([]*PullRequ
 		Find(&prs)
 }
 
+// GetLatestPullRequestsByHeadInfo returns the latest pull request (regardless of its status)
+// by given head information (repo and branch).
+func GetLatestPullRequestsByHeadInfo(repoID int64, branch string) (*PullRequest, error) {
+	prs := make([]*PullRequest, 0, 2)
+	err := x.
+		Where("head_repo_id = ? AND head_branch = ?", repoID, branch).
+		OrderBy("id DESC").
+		Limit(1).
+		Find(&prs)
```

Comment:
Why not just use `Get(&pr)` and remove `Limit(1)`?

Reply:
True... Didn't look at the lines above where a single entry is fetched.
Just adjusted that via the GitHub editor – i.e. I didn't test if it still works (will do later).